joblib/_multiprocessing_helpers.py

```diff
@@ -22,18 +24,34 @@
 #            issue a warning if not
 if mp is not None:
     try:
-        # Use the spawn context
-        if sys.version_info < (3, 3):
-            Semaphore = mp.Semaphore
+        if sys.platform == "win32":
+            # Use the spawn context
+            if PY27:
+                Semaphore = mp.Semaphore
+            else:
+                # Using mp.Semaphore has a border effect and set the default
+                # backend for multiprocessing. To avoid that, we use the
+                # 'spawn' context which is available on all supported platforms
```

Comment:
maybe not mention "all supported platforms" in this comment as this branch of the condition is only for windows so we don't care about the fact that it's available on all other platforms in this comment anymore.

Reply:
Actually, I just re-implemented this to use `_multiprocessing.SemLock`, which can be used on all platform, so we only have one implementation.